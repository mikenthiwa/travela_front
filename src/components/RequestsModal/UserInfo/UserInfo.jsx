import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import ImageLink from '../../image-link/ImageLink';

class UserInfo extends PureComponent {
  render() {
    const { requestData } = this.props;
    return (
      <div className="modal__user-info">
        <ImageLink
          imageSrc={requestData.picture}
          altText="avatar"
          imageClass="modal__oval"
        />
        <div>
          <div>{requestData && requestData.name}</div>
          <div className="user-role">
            {requestData &&
              `${requestData.role || 'N/A'} , 
               ${requestData.department || 'N/A'}`}
          </div>
        </div>
      </div>
    );
  }
}

UserInfo.propTypes = {
  requestData: PropTypes.object
};

UserInfo.defaultProps = {
  requestData: {}
};

export default UserInfo;
