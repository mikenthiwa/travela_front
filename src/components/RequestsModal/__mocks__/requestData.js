const requestData = {
  id: 'xDh20cuGx',
  name: 'Test user C',
  origin: 'Lagos',
  destination: 'Nairobi',
  manager: 'Samuel Kubai',
  department: 'TDD',
  role: 'Software Developer',
  status: 'Approved',
  userId: '-MUyHJmKrxA90lPNQ1FOLNm',
  departureDate: '2018-09-12',
  arrivalDate: '2018-11-12',
  createdAt: '2018-09-04T10:38:12.141Z',
  tripType: 'multi',
  comments: [
    {
      id: 1,
      userName: 'Random User',
      comment: 'This is a comment',
      createdAt: '2018-09-04T10:38:12.141Z',
      picture: 'http://my-image'
    },
    {
      id: 2,
      userName: 'Random User Two',
      comment: 'This is another comment',
      createdAt: '2018-10-04T10:38:12.141Z',
      picture: 'http://my-image'
    },
  ]
};

export default requestData;
