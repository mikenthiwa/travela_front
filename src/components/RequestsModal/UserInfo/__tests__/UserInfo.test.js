import React from 'react';
import { shallow } from 'enzyme';
import UserInfo from '../UserInfo';
import requestData from '../../__mocks__/requestData';

const props = {
  user: {
    picture: 'http://my-image'
  },
  requestData:{
    picture: 'http://my-image'
  }
};

describe('UserComments component', () => {
  const wrapper = mount(<UserInfo {...props} />);

  it('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should render the UserComments component as expected', () => {
    expect(wrapper.find('.modal__user-info').length).toBe(1);
    expect(wrapper.find('.user-role').length).toBe(1);
    expect(wrapper.find('img').length).toBe(1);
    expect(wrapper.length).toBe(1);
  });
});
