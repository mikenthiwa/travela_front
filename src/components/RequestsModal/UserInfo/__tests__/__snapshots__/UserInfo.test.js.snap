// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserComments component should match snapshot 1`] = `
<div
  className="modal__user-info"
>
  <img
    alt="avatar"
    className="modal__oval"
    src="http://my-image"
  />
  <div>
    <div />
    <div
      className="user-role"
    >
      N/A , 
                     N/A
    </div>
  </div>
</div>
`;
